src/planner/binder/statement/bind_update.cpp

```diff
@@ -105,6 +112,11 @@ static void BindUpdateConstraints(TableCatalogEntry &table, LogicalGet &get, Log
 		}
 	}
 
+	// if there is a self referencing foreign key constraint, update_is_del_and_insert sets to false
+	if (is_self_referencing_fk) {
+		update.update_is_del_and_insert = false;
```

Comment:
Updates on indexes are not supported, and by definition a table with foreign keys will have indexes, so this could break things. Is this required? This will require several tests of updates on tables with self-referencing foreign key constraints, together with index look-ups on those tables.